src/firmware/app/control/trajectory_planner.c

```diff
@@ -152,11 +22,15 @@ app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
         variable_time_trajectory->trajectory_elements[0].time;
     constant_period_trajectory->trajectory_elements[0].position =
         variable_time_trajectory->trajectory_elements[0].position;
-    constant_period_trajectory->speed_profile[0] =
-        variable_time_trajectory->speed_profile[0];
-
-    // Keep track of the current time we are searching for in the constant arclength
-    // trajectory
+    constant_period_trajectory->trajectory_elements[0].orientation =
+        variable_time_trajectory->trajectory_elements[0].orientation;
+    constant_period_trajectory->trajectory_elements[0].linear_speed =
+        variable_time_trajectory->trajectory_elements[0].linear_speed;
+    constant_period_trajectory->trajectory_elements[0].angular_speed =
+        variable_time_trajectory->trajectory_elements[0].angular_speed;
```

Comment:
Perhaps it might be worth implementing a `app_position_trajectory_element_copy(PositionTrajectoryElement_t* from, PositionTrajectoryElement_t* to)`? I think that would really help shorten this function, especially if we implement a function below for linear interpolation that outputs a `PositionTrajectoryElement_t` (though I'm not sure if that's abusing the struct or not).

Reply:
Added copy function 👍